lua/damagelogs/shared/autoslay.lua

```diff
@@ -139,181 +139,234 @@ local function CreateCommand()
 
         return
     elseif sam then
-        function sam.autoslay(calling_ply, target, rounds, reason)
-            Damagelog:SetSlays(calling_ply, target:SteamID(), rounds, reason, target)
-        end
+        timer.Simple(1, function()
```

Comment:
Because in SAM the same functions exist in side a zero timer.
We need to override them, Putting them inside a second timer was the easier way to override the existing functions reliably.

Reply:
Just to be clear we could remove the timer and it should work most of the time but I was having issues with functions not getting overridden so I put a timer on it.

SAM: Reason for the timer.
```
	if not GAMEMODE then
		timer.Simple(0, fn)
	else
		fn()
	end
```

Reply:
Ah ok. It's worth adding a comment to explain this next to the timer.